src/components/diagrams/diagram-pane.js

```diff
@@ -0,0 +1,602 @@
+                    diagramView.svgUrl.indexOf(currentNodeRef.current?.id) !==
+                        -1
+                ) {
+                    diagramView.ref?.current?.reloadSvg();
+                }
+            });
+        }
+    }, []);
+
+    // This effect will trigger the diagrams' forced update
+    useEffect(() => {
+        if (studyUpdatedForce.eventData.headers && viewsRef.current) {
+            if (
+                studyUpdatedForce.eventData.headers['updateType'] === 'loadflow'
+            ) {
+                //TODO reload data more intelligently
+                updateDiagram(undefined);
+            } else if (
+                studyUpdatedForce.eventData.headers['updateType'] === 'study'
+            ) {
+                //If the SLD of the deleted substation is open, we close it
+                if (studyUpdatedForce.eventData.headers['deletedEquipmentId']) {
+                    const deletedId =
+                        studyUpdatedForce.eventData.headers[
+                            'deletedEquipmentId'
+                        ];
+                    const vlToClose = viewsRef.current.filter(
+                        (vl) =>
+                            vl.substationId === deletedId || vl.id === deletedId
+                    );
+                    if (vlToClose.length > 0) {
+                        closeDiagramViews([...vlToClose, deletedId]);
+                    }
+
+                    const substationsIds =
+                        studyUpdatedForce.eventData.headers['substationsIds'];
+                    viewsRef.current.forEach((v) => {
+                        const vl = network.getVoltageLevel(v.id);
+                        if (vl && substationsIds.includes(vl.substationId)) {
+                            updateDiagram(vl.id);
+                        }
+                    });
+                } else {
+                    updateDiagram(undefined);
+                }
+            } else if (
+                studyUpdatedForce.eventData.headers['updateType'] ===
+                'buildCompleted'
+            ) {
+                if (
+                    studyUpdatedForce.eventData.headers['node'] ===
+                    currentNodeRef.current?.id
+                ) {
+                    updateDiagram(undefined);
+                }
+            }
+        }
+        // Note: studyUuid, and loadNetwork don't change
+    }, [
+        studyUpdatedForce,
+        dispatch,
+        studyUuid,
+        updateDiagram,
+        closeDiagramViews,
+        network,
+    ]);
+
+    /**
+     * DIAGRAM SIZE COMPUTATION
+     */
+
+    const [computedHeight, setComputedHeight] = useState();
+
+    useEffect(() => {
+        let displayedDiagramHeights_ =
+            displayedDiagramHeightsRef.current?.filter((displayedHeight) =>
+                views
+                    .filter((sld) => sld.state !== ViewState.MINIMIZED)
+                    .map((sld) => sld.id)
```

Comment:
Here, we'll have the same id for the SLD and the NAD of a same voltage level

Reply:
Thanks, it is now fixed.